packages/mermaid/src/diagrams/git/gitGraphAst.js

```diff
@@ -274,21 +275,33 @@ export const cherryPick = function (sourceId, targetId, tag) {
     };
     throw error;
   }
-
   let sourceCommit = commits[sourceId];
   let sourceCommitBranch = sourceCommit.branch;
   if (sourceCommit.type === commitType.MERGE) {
-    let error = new Error(
-      'Incorrect usage of "cherryPick". Source commit should not be a merge commit'
-    );
-    error.hash = {
-      text: 'cherryPick ' + sourceId + ' ' + targetId,
-      token: 'cherryPick ' + sourceId + ' ' + targetId,
-      line: '1',
-      loc: { first_line: 1, last_line: 1, first_column: 1, last_column: 1 },
-      expected: ['cherry-pick abc'],
-    };
-    throw error;
+    if (!parentCommitId) {
+      let error = new Error(
+        'Incorrect usage of cherry-pick: If the source commit is a merge commit, an immediate parent commit must be specified.'
+      );
+      error.hash = {
+        text: `cherryPick ${sourceId} ${targetId}`,
+        token: `cherryPick ${sourceId} ${targetId}`,
+        line: '1',
+        loc: { first_line: 1, last_line: 1, first_column: 1, last_column: 1 },
```

Comment:
If it's not possible to get the correct values, then it's better to avoid putting the hash.